packages/ragbits-core/src/ragbits/core/audit/cli_span.py

```diff
@@ -0,0 +1,78 @@
+import time
+from typing import Optional
+
+from rich.tree import Tree
+
+
+class CLISpan:
+    """
+    CLI Span represents a single operation within a trace.
+    """
+
+    def __init__(self, name: str, parent: Optional["CLISpan"] = None):
+        """
+        Constructs a new CLI Span.
+        Sets the start time of the span - the wall time at which the operation started.
+        Sets the span status to 'started'.
+
+        Args:
+            name: The name of the span.
+            parent: the parent of initiated span.
+        """
+        self.name = name
+        self.parent = parent
+        self.start_time: float = time.time()
+        self.end_time: float | None = None
+        self.children: list[CLISpan] = []
+        self.status: str = "started"
+
+    def end(self) -> None:
+        """Sets the current time as the span's end time.
+        The span's end time is the wall time at which the operation finished.
+        Only the first call to `end` should modify the span,
+        further calls are ignored.
+        """
+        if self.end_time is None:
+            self.end_time = time.time()
+
+    def to_dict(self) -> dict:
+        """
+        Convert the CLISpan object and its children into a dictionary representation.
+
+        Returns:
+        dict: A dictionary containing the span's name, start time, end time, events,
+              and a list of its children's dictionary representations.
+        """
+        return {
+            "name": self.name,
+            "start_time": self.start_time,
+            "end_time": self.end_time,
+            "children": [child.to_dict() for child in self.children],
+            "parent": self.parent.name if self.parent else None,
+        }
+
+    def to_tree(self, tree: Tree | None = None, color: str = "bold blue") -> Tree | None:
```

Comment:
`to_tree` name - to me suggests construction of a new object from the class instance

Reply:
Could you give me better name?